src/main/java/com/cinema/security/CustomUserDetailService.java

```diff
@@ -0,0 +1,32 @@
+package com.cinema.security;
+
+import com.cinema.model.User;
+import com.cinema.service.UserService;
+import java.util.stream.Collectors;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailService implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailService(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        User user = userService.findByEmail(email).orElseThrow();
```

Comment:
we can throw UsernameNotFoundException() here

Reply:
We can - I did but either I did smth wrong but I don't get a right message when exception is thrown...